diff --git a/pappl-retrofit/pappl-retrofit.c b/pappl-retrofit/pappl-retrofit.c
index dfbf570..50f785b 100644
--- a/pappl-retrofit/pappl-retrofit.c
+++ b/pappl-retrofit/pappl-retrofit.c
@@ -73,6 +73,22 @@ prRetroFitPrinterApp(pr_printer_app_config_t *printer_app_config,
   cupsArrayDelete(global_data.config->stream_formats);
   if (global_data.config->driver_selection_regex_list)
     cupsArrayDelete(global_data.config->driver_selection_regex_list);
+
+  if (global_data.drivers)
+  {
+    for (int i = 0; i < global_data.num_drivers; i++)
+    {
+      free((char*)global_data.drivers[i].device_id);
+      free((char*)global_data.drivers[i].name);
+      free((char*)global_data.drivers[i].description);
+      free((char*)global_data.drivers[i].extension);
+    }
+
+    free(global_data.drivers);
+  }
+
+  if (global_data.ppd_paths)
+    cupsArrayDelete(global_data.ppd_paths);
   
   return (ret);
 }
